extensions/control-plane/store/sql/asset-index-sql/src/main/java/org/eclipse/dataspaceconnector/sql/assetindex/SqlAssetIndex.java

```diff
@@ -152,44 +161,48 @@ public void accept(AssetEntry item) {
                 throw new EdcPersistenceException(e);
             }
         });
-
-
     }
 
     @Override
     public Asset deleteById(String assetId) {
         Objects.requireNonNull(assetId);
 
-        try (var connection = getConnection()) {
-            var asset = findById(assetId);
-            if (asset == null) {
-                return null;
-            }
+        return transactionContext.execute(() -> {
+            try (var connection = getConnection()) {
+                var asset = findById(assetId);
+                if (asset == null) {
+                    return null;
+                }
 
```

Comment:
please do not remove the transaction block, as there may be a transaction already in progress, to which the delete statement is attached.

Reply:
@paullatzelsperger actual I didn't removed, but I "enlarged" to contain both the `findBy` and the `delete` queries, as they should stay in the same transaction context, and that was not the case.